coverage/hare_coverage.py

```diff
@@ -0,0 +1,374 @@
+            print(f"Error {e} while cleanup.")
+
+    def check_prerequisite(self) -> None:
+        """
+        install coverage related packages if not present
+        :return: None
+        """
+        Utility.print_topic("Prerequisite Checking.")
+        PipInstaller(["coverage", "pytest", "pytest-cov"])
+
+    def run_ut(self) -> None:
+        """
+        Execute unit testcases.
+        :return: None
+        """
+        Utility.print_topic("Running UT's.")
+        for test_dir_path in Utility.search_dir(getcwd(), "test"):
+            out, _ = Utility.run_cmd(f"pytest {test_dir_path}")
+            execution_logs: str = out.decode("utf-8")
+            if search("ERROR", execution_logs, IGNORECASE):
+                print(f"Error in {test_dir_path!r} unittest execution.\n")
+                print(execution_logs, "\n")
+
+    def run_coverage(self) -> None:
+        """
+        Running coverage tool over UTs.
+        :return: None
+        """
+        Utility.print_topic("Running Coverage.")
+        command_line: str
+        for index, (src, tst) in enumerate(self.src_tst_map.items()):
+            if index == 0:
+                command_line: str = f"pytest --cov={src} --cov-report={self.cov_type}:{self.report} {tst}"
+            else:
+                command_line: str = f"pytest --cov={src} --cov-report={self.cov_type}:{self.report} --cov-append {tst}"
+            try:
+                Utility.run_cmd(command_line)
+            except Exception as e:
+                print(f"Error {e} while executing {command_line!r}")
+
+    def check_report(self) -> None:
+        """
+        Validate the report generated.
+        :return: None
+        """
+        Utility.print_topic("Checking report.")
+        if Utility.is_exist(self.report) is True:
+            print(f"Python Code Coverage Report generated successfully at {self.report!a}.")
+        else:
+            print("Failed to generate the Coverage report.")
+
+
+class CCoverage(Coverage):
+
+    def cleanup(self) -> None:
+        """
+        Cleaning up old coverage reports.
+        :return: None
+        """
+        pass
+
+    def check_prerequisite(self) -> None:
+        """
+        install coverage package if not present
+        :return: None
+        """
+        pass
+
+    def run_ut(self) -> None:
+        """
+        Execute unit testcases.
+        :return: None
+        """
+        pass
+
+    def run_coverage(self) -> None:
+        """
+        Running coverage tool over UTs.
+        :return: None
+        """
+        pass
+
+    def check_report(self) -> None:
+        """
+        Validate the report generated
+        :return: None
+        """
+        pass
+
+
+class Executor:
+
+    """
+    This class is responsible for the execution of the coverage for both
+    python and C.
+    """
+
+    def __init__(self, cov: Coverage):
+        """
+        Initialization.
+        :param cov: Coverage
+        """
+        Coverage.setup_env()

```

Comment:
Invoking the functions from the name of the class destroys all the benefits that OOP could bring. Please avoid it.

Reply:
ok